connect/runtime/src/main/java/org/apache/kafka/connect/runtime/SourceTaskOffsetCommitter.java

```diff
@@ -65,72 +76,46 @@ public void close(long timeoutMs) {
     }
 
     public void schedule(final ConnectorTaskId id, final WorkerSourceTask workerTask) {
-        synchronized (committers) {
-            long commitIntervalMs = config.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
-            ScheduledFuture<?> commitFuture = commitExecutorService.schedule(new Runnable() {
-                @Override
-                public void run() {
-                    commit(id, workerTask);
-                }
-            }, commitIntervalMs, TimeUnit.MILLISECONDS);
-            committers.put(id, new ScheduledCommitTask(commitFuture));
-        }
+        long commitIntervalMs = config.getLong(WorkerConfig.OFFSET_COMMIT_INTERVAL_MS_CONFIG);
+        ScheduledFuture<?> commitFuture = commitExecutorService.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                commit(workerTask);
+            }
+        }, 0, commitIntervalMs, TimeUnit.MILLISECONDS);
+        committers.put(id, commitFuture);
     }
 
     public void remove(ConnectorTaskId id) {
-        final ScheduledCommitTask task;
-        synchronized (committers) {
-            task = committers.remove(id);
-            task.cancelled = true;
-            task.commitFuture.cancel(false);
-        }
-        if (task.finishedLatch != null) {
-            try {
-                task.finishedLatch.await();
-            } catch (InterruptedException e) {
-                throw new ConnectException("Unexpected interruption in SourceTaskOffsetCommitter.", e);
-            }
-        }
-    }
+        final ScheduledFuture<?> task = committers.remove(id);
+        if (task == null)
```

Comment:
While not expected in practice (another way to say it's not possible), I see this check as a safe guard w.r.t to the use of this concurrent collection. There's no way of putting a `null` value since `scheduleWithFixedDelay` always returns a future object. 

But given that this `remove` method is currently called in at least two places (by potentially different threads) (via `stopTask` in `stopAndAwaitTask` and in `stopTasks`) I wouldn't mind leaving this as a fail-safe option that doesn't hurt to have here (An NPE here will be rare but also weird and not very informative).